include/mbedtls/ccm.h

```diff
@@ -118,6 +116,10 @@ int mbedtls_ccm_encrypt_and_tag( mbedtls_ccm_context *ctx, size_t length,
  *
  * \return         0 if successful and authenticated,
  *                 MBEDTLS_ERR_CCM_AUTH_FAILED if tag does not match
+ *
+ * \note           MBEDTLS_ERR_CCM_AUTH_FAILED should NOT be returned on
+ *                 fatal internal errors, since it is remapped to
```

Comment:
@RonEld I'd prefer not to explain any internals here - perhaps we decide to handle the error code differently in the future. What about something like "\note The error code MBEDTLS_ERR_CCM_AUTH_FAILED is for tag mismatches only and MUST NOT be used for fatal internal errors." Also, even if it's a slight duplication, we could also enhance the \return section, e.g. "... MBEDTLS_ERR_CCM_AUTH_FAILED if tag does not match, or any other error code for a fatal internal error."